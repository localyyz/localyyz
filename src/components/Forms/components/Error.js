import React from "react";
import { View, StyleSheet, Text } from "react-native";

// custom
import { Styles, Colours } from "localyyz/constants";

export default class Error extends React.Component {
  render() {
    return (
      <View style={styles.container}>
        <Text numberOfLines={1} style={styles.label}>
          {this.props.children}
        </Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {},

  label: {
    ...Styles.Text,
    ...Styles.SmallText,
    color: Colours.Fail
  }
});
