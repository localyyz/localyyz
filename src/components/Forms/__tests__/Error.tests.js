import React from "react";
import Renderer from "react-test-renderer";

// local
import Error from "../components/Error";

// constants
const NAME = "Error Test - Form";
const PROPS = {
  children: "sample chidren"
};

const Component = Error;

describe(NAME, () => {
  it(`${NAME}: should render properly`, () => {
    let c = Renderer.create(<Component {...PROPS} />).getInstance();
    expect(c).not.toBeUndefined();
  });

  it(`${NAME}: should render to snapshot`, () => {
    expect(
      Renderer.create(<Component {...PROPS} />).toJSON()
    ).toMatchSnapshot();
  });
});
