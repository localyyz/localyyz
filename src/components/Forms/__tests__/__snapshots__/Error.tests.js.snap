// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Test - Form Error Test - Form: should render to snapshot 1`] = `
<View
  style={Object {}}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Object {
        "backgroundColor": "rgba(255, 255, 255, 0)",
        "color": "#FF0000",
        "fontSize": 14,
        "fontWeight": "300",
      }
    }
  >
    sample chidren
  </Text>
</View>
`;
